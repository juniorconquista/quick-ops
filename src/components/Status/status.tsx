import React, { memo } from 'react';
import { Box, H1, H2 } from 'quarks-ui';

const Status: React.FC = () => (
    <Box
        styling="column"
        minHeight="86px"
        width="80%"
        minWidth="250px"
        border="1px solid"
        borderRadius="6px"
        borderColor="#009d43"
        bg="#edfff5"
        p="0px 32px"
    >
        <H1 styling="light" color="#009d43" fontSize="16px" mb="8px">
            Status
        </H1>
        <H2 styling="base" color="#009d43" fontSize="20px" fontWeight="bold">
            Conclu√≠do
        </H2>
    </Box>
);

export default memo(Status);
