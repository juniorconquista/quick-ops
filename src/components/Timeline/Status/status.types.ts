import { ReactNode } from 'react';

export type StatusStyledProps = {
    /** Each styling variant has unique properties that are inherited through the type informed */
    styling?: 'base' | 'PENDING' | 'APPROVED' | 'REFUSED';
    children?: ReactNode;
};
