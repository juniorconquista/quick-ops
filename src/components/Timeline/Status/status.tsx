import React, { memo } from 'react';
import { variant, VariantArgs } from 'styled-system';
import styled from 'styled-components';

import { StatusStyledProps } from './status.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            height: '24px',
            borderRadius: '18px',
            border: '1px solid #6b7480',
            color: '#6b7480',
            fontSize: '12px',
            fontWeight: 'bold',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            bg: '#f0f3f7',
            p: '0px 20px',
        },
        get PENDING() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
            };
        },
        get REFUSED() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#fde7e7',
                borderColor: '#bd0f0f',
                color: '#bd0f0f',
            };
        },
        get APPROVED() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#e6fff0',
                borderColor: '#03a046',
                color: '#03a046',
            };
        },
    },
} as VariantArgs);

const Status: React.FC<StatusStyledProps> = styled.div<StatusStyledProps>`
    ${variants}
`;

export default memo(Status);
