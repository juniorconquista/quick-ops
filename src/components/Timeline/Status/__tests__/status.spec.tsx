import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import ThemeProvider from '../../../../utils/ThemeProviderTest';

import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

import Status from '../status';

describe('Status component', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });
    test('basic snapshot', () => {
        const container = render(
            <ThemeProvider>
                <Status styling="base" />
            </ThemeProvider>,
        );
        expect(container.asFragment()).toMatchSnapshot();
    });

    it('should inherit the styling properties of the Pending variant', () => {
        const status = renderer
            .create(
                <ThemeProvider>
                    <Status styling="PENDING" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(status).toHaveStyleRule('border', '1px solid #6b7480');
        expect(status).toHaveStyleRule('color', '#6b7480');
        expect(status).toHaveStyleRule('background-color', '#f0f3f7');
    });

    it('should inherit the styling properties of the Refused variant', () => {
        const status = renderer
            .create(
                <ThemeProvider>
                    <Status styling="REFUSED" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(status).toHaveStyleRule('border-color', '#bd0f0f');
        expect(status).toHaveStyleRule('color', '#bd0f0f');
        expect(status).toHaveStyleRule('background-color', '#fde7e7');
    });

    it('should inherit the styling properties of the Approved variant', () => {
        const status = renderer
            .create(
                <ThemeProvider>
                    <Status styling="APPROVED" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(status).toHaveStyleRule('border-color', '#03a046');
        expect(status).toHaveStyleRule('color', '#03a046');
        expect(status).toHaveStyleRule('background-color', '#e6fff0');
    });
});
