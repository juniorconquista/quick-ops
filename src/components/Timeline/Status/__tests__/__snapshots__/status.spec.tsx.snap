// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component basic snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 24px;
  border-radius: 18px;
  border: 1px solid #6b7480;
  color: #6b7480;
  font-size: 12px;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #f0f3f7;
  padding: 0px 20px;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
